src/shared_realm.cpp

```diff
@@ -33,9 +33,28 @@
 #include <realm/history.hpp>
 #include <realm/util/scope_exit.hpp>
 
+#include <sstream>
+
 using namespace realm;
 using namespace realm::_impl;
 
+static std::string get_initial_named_pipe_directory()
+{
+    auto tmp_dir = getenv("TMPDIR");
+    if (!tmp_dir) {
+        return std::string();
+    }
+    std::string tmp_dir_str(tmp_dir);
+    if (tmp_dir_str.back() == '/') {
+        return tmp_dir_str;
+    }
+    std::ostringstream ss;
```

Comment:
There's no reason to use a stringstream just to append a character; just do `tmp_dir_str += '/';`.